tutorials/text/gliner-pii-redaction/gliner_pii_redactor.py

```diff
@@ -0,0 +1,186 @@
+
+from nemo_curator.backends.base import NodeInfo, WorkerMetadata
+from nemo_curator.stages.base import ProcessingStage
+from nemo_curator.stages.resources import Resources
+from nemo_curator.tasks import DocumentBatch
+
+# Model identifier from Hugging Face
+GLINER_PII_MODEL_IDENTIFIER = "nvidia/gliner-pii"
+# Full list of entities and quasi identifiers
+PII_LABELS = [
+    "first_name",
+    "last_name",
+    "name",
+    "street_address",
+    "city",
+    "state",
+    "postcode",
+    "country",
+    "address",
+    "latitude",
+    "longitude",
+    "coordinate",
+    "age",
+    "phone_number",
+    "fax_number",
+    "email",
+    "ssn",
+    "unique_identifier",
+    "medical_record_number",
+    "health_plan_beneficiary_number",
+    "account_number",
+    "certificate_license_number",
+    "vehicle_identifier",
+    "license_plate",
+    "device_identifier",
+    "biometric_identifier",
+    "url",
+    "ipv4",
+    "ipv6",
+    "national_id",
+    "tax_id",
+    "bank_routing_number",
+    "swift_bic",
+    "credit_debit_card",
+    "cvv",
+    "pin",
+    "employee_id",
+    "api_key",
+    "customer_id",
+    "user_name",
+    "password",
+    "mac_address",
+    "http_cookie",
+    "date",
+    "date_time",
+    "blood_type",
+    "gender",
+    "sexuality",
+    "political_view",
+    "race",
+    "ethnicity",
+    "religious_belief",
+    "language",
+    "education",
+    "job_title",
+    "employment_status",
+    "company_name",
+]
+
+
+@dataclass
+class GlinerPiiRedactor(ProcessingStage[DocumentBatch, DocumentBatch]):
+    """
+    The module responsible for redacting PII entities from text.
+    It accepts a list of labels to redact and a threshold for the confidence score.
+
+    Args:
+        text_field (str): The field to read the text from. Default is "text".
+        labels (list[str]): The labels to redact.
+        threshold (float): The threshold for the confidence score. Default is 0.5.
+        use_gpu (bool): Whether to use GPU acceleration. Default is True.
+        model_inference_batch_size (int): The batch size for model inference. Default is 128.
+
+    """
+
+    text_field: str = "text"
+    labels: list[str] | None = None
+    threshold: float = 0.5
+    use_gpu: bool = True
+    model_inference_batch_size: int = 128
+    _name: str = "gliner_pii_redactor"
+
+    def __post_init__(self):
+        if self.labels is None:
+            self.labels = PII_LABELS
+        if self.use_gpu:
+            self._resources = Resources(cpus=1, gpus=1)
+            self.device = "cuda"
+        else:
+            self._resources = Resources(cpus=1)
+            self.device = "cpu"
+
+    def inputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], [self.text_field]
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        return ["data"], [self.text_field]
+
+    def ray_stage_spec(self) -> dict[str, Any]:
+        return {"is_actor_stage": True}
+
+    def setup_on_node(
```

Comment:
I noticed both image and video do models under the ModelInterface, while on text side we do directly on the Model.. We should probably consider consolidating (not in this PR) and create a ticket for the same